go/oasis-test-runner/scenario/e2e/e2e.go

```diff
@@ -0,0 +1,114 @@
+// Package e2e implements the Oasis e2e test scenarios.
+package e2e
+
+import (
+	"fmt"
+	"math"
+	"os/exec"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	"github.com/oasislabs/oasis-core/go/oasis-test-runner/env"
+	"github.com/oasislabs/oasis-core/go/oasis-test-runner/oasis"
+)
+
+const (
+	cfgNodeBinary = "node.binary"
+	cfgHaltEpoch  = "halt_epoch"
+)
+
+var (
+	// E2eParamsDummy is a dummy instance of e2eImpl used to register e2e-wise parameters.
+	E2eParamsDummy *e2eImpl = &e2eImpl{name: "e2e"}
+
+	logger = logging.GetLogger("e2e/common")
+)
+
+type e2eImpl struct {
+	net    *oasis.Network
+	name   string
+	logger *logging.Logger
+
+	// nodeBinary is the path to oasis-node executable.
+	nodeBinary string
+
+	// haltEpoch is halt epoch height.
+	haltEpoch uint64
+}
+
+func newE2eImpl(name string) *e2eImpl {
+	return &e2eImpl{
+		name:       "e2e/" + name,
+		logger:     logging.GetLogger("scenario/e2e/" + name),
+		nodeBinary: "oasis-node",
+		haltEpoch:  math.MaxUint64,
+	}
+}
+
+func (sc *e2eImpl) Clone() e2eImpl {
+	return e2eImpl{
+		net:        sc.net,
+		name:       sc.name,
+		logger:     sc.logger,
+		nodeBinary: sc.nodeBinary,
+		haltEpoch:  sc.haltEpoch,
+	}
+}
+
+func (sc *e2eImpl) Name() string {
+	return sc.name
+}
+
+func (sc *e2eImpl) Parameters() *flag.FlagSet {
+	fs := flag.NewFlagSet(sc.name, flag.ContinueOnError)
+	fs.StringVar(&sc.nodeBinary, cfgNodeBinary, sc.nodeBinary, "path to the node binary")
+	fs.Uint64Var(&sc.haltEpoch, cfgHaltEpoch, sc.haltEpoch, "halt epoch height")
+
+	return fs
+}
+
+func (sc *e2eImpl) Init(childEnv *env.Env, net *oasis.Network) error {
+	sc.net = net
+	return nil
+}
+
+func startClient(env *env.Env, net *oasis.Network, binary string, clientArgs []string) (*exec.Cmd, error) {
```

Comment:
Should this be changed to be a method on `runtimeImpl` instead? Then you can get net, binary and clientArgs from it.